client/benches/torii.rs

```diff
@@ -35,14 +35,14 @@ fn query_requests(criterion: &mut Criterion) {
         &configuration.genesis,
         &configuration.sumeragi.transaction_limits,
     )
-    .expect("genesis creation failed");
+    .expect("genesis creation failed")
+    .expect("genesis created");
 
-    rt.block_on(peer.start_with_config_permissions(
-        configuration.clone(),
-        genesis,
-        AllowAll,
-        AllowAll,
-    ));
+    let builder = PeerBuilder::new()
+        .with_configuration(configuration.clone())
+        .with_genesis(genesis);
+
+    rt.block_on(builder.start_with_peer(&mut peer));
```

Comment:
That's why I think this design is a bit ugly.

Lets think about an alternative design. So, we have several facts:
1. Instance of `Peer` can starts and also restarts.
2. `Peer` uses only values (which keep inside `PeerBuilder`) when calling the `start` method.
3. Instance `Peer` doesn't means "an already running peer", that is, instance of `Peer` can be pre-created to keep inside only the ID (and other parameters) and can be starts later.

But the alternative design is to introduce a `build` method for `PeerBuilder` and to pass the values of `PeerBuilder` to `Peer` when calling the `build`. So `Peer` will have to store all these values internally and will look like this:
```rust
struct Peer {
    /// id of peer
    pub id: PeerId,
    /// api address
    pub api_address: String,
    /// bla bla bla ....
   
    /// ... and also these parameters
    genesis: WithGenesis<G>,
    configuration: Option<Configuration>,
    instruction_validator: Option<IsInstructionAllowedBoxed<W>>,
    query_validator: Option<IsQueryAllowedBoxed<W>>,
    temp_dir: Option<Arc<TempDir>>,
}

/// ....

impl PeerBuilder {
    /// ...
    fn build(self) -> Peer {
        Peer::from(self.genesis, self.configuration, self.instruction_validator, // ...)
    }
}
```

Reply:
I've just realized that I can rename `PeerBuilder` to `PeerArgsBuilder` or something. And make the `Peer::start` method accept the `PeerArgsBuilder` as argument. It will look like this:
```rust
Peer::new().start(PeerArgsBuilder::new().with_configuration(config).build()).await;
```

This is the same idea but more pretty interface.